sky/serve/replica_managers.py

```diff
@@ -257,12 +266,14 @@ def to_replica_status(self) -> serve_state.ReplicaStatus:
 class ReplicaInfo:
     """Replica info for each replica."""
 
-    def __init__(self, replica_id: int, cluster_name: str) -> None:
+    def __init__(self, replica_id: int, cluster_name: str,
+                 use_spot: bool) -> None:
         self.replica_id: int = replica_id
         self.cluster_name: str = cluster_name
         self.first_not_ready_time: Optional[float] = None
         self.consecutive_failure_times: List[float] = []
         self.status_property: ReplicaStatusProperty = ReplicaStatusProperty()
+        self.use_spot: bool = use_spot
```

Comment:
Since now we are peeking at the `launched_resources`, do we still need these `use_spot`s?

Reply:
ditto

Reply:
let me remove them